clicommand/pipeline_upload.go

```diff
@@ -174,6 +181,30 @@ var PipelineUploadCommand = cli.Command{
 			logger.Fatal("Pipeline parsing of \"%s\" failed (%s)", filename, err)
 		}
 
+		// In dry-run mode we just output the generated pipeline to stdout
+		if cfg.DryRun {
+			enc := json.NewEncoder(os.Stdout)
+			enc.SetIndent("", "  ")
+
+			// Dump json indented to stdout. All logging happens to stderr
+			// this can be used with other tools to get interpolated json
+			if err := enc.Encode(parsed); err != nil {
+				logger.Fatal("%#v", err)
+			}
+
+			os.Exit(0)
+		}
+
+		// Check we have a job id set if not in dry run
+		if cfg.Job == "" {
+			logger.Fatal("Missing job parameter. Usually this is set in the environment for a Buildkite job via BUILDKITE_JOB_ID.")
```

Comment:
I'd love a way to make more informative error messages like this the standard vs the standard "required parameter" message.